### ISSUE TITLE
Add JSON Functions to API and JDBC Adapter

### ISSUE THREAD

**Issue by exanm** [exasol/virtual-schemas#324]
Please add support for the new JSON functions to the API and the JDBC adapter.

### Example Queries

```sql
SELECT '{"a": 1}' IS JSON VALUE WITHOUT UNIQUE KEYS FROM dual; -- returns true
SELECT 'not JSON' IS JSON VALUE WITHOUT UNIQUE KEYS FROM dual; -- returns false
```
With `IS NOT JSON` the result of those queries would be negated.

```sql
SELECT JSON_VALUE('{"a": 1}', '$.a' RETURNING VARCHAR(100) UTF8 NULL ON EMPTY NULL ON ERROR) FROM dual; -- returns '1'
```

### Capabilities

New capabilities:

* `FN_PRED_IS_JSON`
* `FN_PRED_IS_NOT_JSON`
* `FN_JSON_VALUE`

### Predicates

New predicates:

`exp1 IS JSON {VALUE | ARRAY | OBJECT | SCALAR} {WITH | WITHOUT} UNIQUE KEYS` (requires predicate capability `IS_JSON`)
```json
{
    "type": "predicate_is_json",
    "expression": {
        ...
    },
    "typeConstraint": "VALUE",
    "keyUniquenessConstraint": "WITHOUT UNIQUE KEYS"
}
```

The same can be used for `predicate_is_not_json` (requires predicate capability `IS_NOT_JSON`).

Notes:
* `typeConstraint` is `"VALUE"`, `"ARRAY"`, `"OBJECT"`, or `"SCALAR"`.
* `keyUniquenessConstraint` is `"WITH UNIQUE KEYS"` or `"WITHOUT UNIQUE KEYS"`.

### Scalar Functions

New special case for scalar functions:

`JSON_VALUE(arg1, arg2 RETURNING dataType {ERROR | NULL | DEFAULT exp1} ON EMPTY {ERROR | NULL | DEFAULT exp2} ON ERROR)` (requires scalar-function capability `JSON_VALUE`)
```json
{
    "type": "function_scalar_json_value",
    "name": "JSON_VALUE",
    "arguments":
    [
        {
            ...
        },
        {
            ...
        }
    ],
    "returningDataType": dataType,
    "emptyBehavior":
    {
        "type": "ERROR"
    },
    "errorBehavior":
    {
        "type": "DEFAULT",
        "expression": exp2
    }
}
```

Notes:
* `arguments`: Contains two entries: The JSON item and the path specification.
* `emptyBehavior` and `errorBehavior`: `type` is `"ERROR"`, `"NULL"`, or `"DEFAULT"`. Only for `"DEFAULT"` the member `expression` containing the default value exists.

**Comment by AnastasiiaSergienko** [comment]
The functions were implemented in 
https://github.com/exasol/virtual-schema-common-java/issues/131

Here I'm adding the new API description to the documentation.

**Comment by AnastasiiaSergienko** [comment]
Closed with #331
